src/main.cpp

```diff
@@ -3979,7 +3979,6 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             mempool.check(pcoinsTip);
             RelayTransaction(tx);
             vWorkQueue.push_back(inv.hash);
-            vEraseQueue.push_back(inv.hash);
```

Comment:
This tx isn't in the orphan map.   We're just now receiving it off the network and accepting it to the mempool, so it never entered the orphan map.  If it was a duplicate transaction that was an orphan the first time it was received, it would have previously been accepted to the mempool at the time its inputs were satisfied, and wouldn't be accepted here.

It doesn't hurt to add tx to vEraseQueue, but its unnecessary and confuses the logic.